src/controllers/LocalStorageHistoryController.ts

```diff
@@ -55,26 +56,52 @@ export class LocalStorageHistoryController extends RootComponent implements IHis
     this.omit = attributes;
   }
 
+  public setState(values: Record<string, any>) {
+    this.storage.save(values);
+  }
+
   private updateLocalStorageFromModel() {
-    const attributes = _.omit(this.model.getAttributes(), this.omit);
+    const attributes = omit(this.model.getAttributes(), this.omit);
     this.setState(attributes);
     this.logger.debug('Saving state to localstorage', attributes);
   }
 
-  private updateModelFromLocalStorage() {
-    const toSet: { [key: string]: any } = {};
-    const loadedFromStorage = this.storage.load();
-    _.each(<_.Dictionary<any>>this.model.attributes, (value, key?, obj?) => {
-      var valToSet = loadedFromStorage ? loadedFromStorage[key] : undefined;
-      if (valToSet == undefined) {
-        valToSet = this.model.defaultAttributes[key];
-      }
-      toSet[key] = valToSet;
+  private initModelFromLocalStorageAndUrlHash() {
+    const model = {
+      ...this.localStorageModel,
+      ...this.urlHashModel
+    };
+
+    this.model.setMultiple(model);
+  }
+
+  private get localStorageModel() {
+    const model: Record<string, any> = {};
+    const storedValues = this.storage.load() || {};
+
+    each(this.model.attributes, (value, key) => {
+      const storedValue = storedValues[key];
+      const defaultValue = this.model.defaultAttributes[key];
+      const valueToSet = storedValue == undefined ? defaultValue : storedValue;
+
+      model[key] = valueToSet;
     });
-    this.model.setMultiple(toSet);
+
+    return model;
   }
 
-  public setState(values: Record<string, any>) {
-    this.storage.save(values);
+  private get urlHashModel(): Record<string, any> {
```

Comment:
Is specifying the return type necessary when you return `model` that as the same type? Ts should automatically define it well